libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -96,24 +96,29 @@ internal class MapboxTripSession(
                 setRouteToNativeNavigator(routes, legIndex)
             }
             RoutesExtra.ROUTES_UPDATE_REASON_ALTERNATIVE -> {
-                NativeSetRouteResult(
+                NativeSetRouteValue(
                     nativeAlternatives = navigator.setAlternativeRoutes(routes.drop(1))
                 )
             }
             RoutesExtra.ROUTES_UPDATE_REASON_REFRESH -> {
-                val alternatives = if (routes.isNotEmpty()) {
+                if (routes.isNotEmpty()) {
                     val primaryRoute = routes.first()
                     val alternatives = navigator.refreshRoute(primaryRoute)
                     roadObjects = getRouteInitInfo(primaryRoute.nativeRoute().routeInfo)
                         ?.roadObjects
                         ?: emptyList()
                     this@MapboxTripSession.primaryRoute = routes.first()
-                    alternatives
+                    if (alternatives != null) {
```

Comment:
it's not obvious that null means error. Maybe it would make sense to return an `Expected` from `com.mapbox.navigation.navigator.internal.MapboxNativeNavigatorImpl#refreshRoute`?